ibridges/irodsconnector/data_operations.py

```diff
@@ -1,36 +1,35 @@
 """ collections and data objects
 """
-import logging
 from typing import Optional, Union
+from pathlib import Path
+import os
+import warnings
 
 import irods.collection
 import irods.data_object
 import irods.exception
+from irods.models import DataObject
 
 from ibridges.irodsconnector import keywords as kw
-from ibridges.irodsconnector.resources import Resources
 from ibridges.irodsconnector.session import Session
-from ibridges.utils.path import IrodsPath, LocalPath
+from ibridges.utils.path import IrodsPath
 
 
-class DataOperation():
+class DataOperations():
     """ Irods collections and data objects operations"""
 
-    def __init__(self, resources: Resources, session: Session):
+    def __init__(self, session: Session):
         """ iRODS data operations initialization
 
             Parameters
             ----------
-            resources : Resource
-                Instance of the Resource class
             session : Session
                 instance of the Session class
 
         """
-        self.resources = resources
         self.session = session
 
-    def get_dataobject(self, path: str) -> irods.data_object.iRODSDataObject:
+    def get_dataobject(self, path: Union[str, IrodsPath]) -> irods.data_object.iRODSDataObject:
```

Comment:
For now, I would leave it in here. In the python API this function belongs to the data_objects module and it returns a python object of class data_object, while I think it would also be logical to let the IrodsPath only return IrodsPaths, to have a distinct difference between string operations with occasional validation with the Irods server and real operations on the data objects and collections.
For now I would leave it like this and move this discussion to the discussion about our class structure, see below.